aws_tools/s3_handler.py

```diff
@@ -39,15 +39,14 @@ def download_file(self, key, local_file):
     # Downloads all the files in S3 that have a prefix of `key_prefix` from `bucket` to the `local` directory
     def download_dir(self, key_prefix, local):
         paginator = self.client.get_paginator('list_objects')
-        for result in paginator.paginate(Bucket=self.bucket_name, Delimiter='/', Prefix=key_prefix):
-            if result.get('CommonPrefixes') is not None:
-                for subdir in result.get('CommonPrefixes'):
-                    self.download_dir(subdir.get('Prefix'), local)
-            if result.get('Contents') is not None:
-                for f in result.get('Contents'):
-                    if not os.path.exists(os.path.dirname(local + os.sep + f.get('Key'))):
-                        os.makedirs(os.path.dirname(local + os.sep + f.get('Key')))
-                    self.download_file(f.get('Key'), local + os.sep + f.get('Key'))
+        result = paginator.paginate(Bucket=self.bucket_name, Delimiter='/', Prefix=key_prefix)
```

Comment:
The previous code didn't seem to line up with the docs for `paginate(..)` ([see here](https://boto3.readthedocs.io/en/latest/reference/services/s3.html#S3.Paginator.ListObjectVersions.paginate))